Okta/OktaAuth.swift

```diff
@@ -60,6 +62,46 @@ public struct OktaAuthorization {
     func passwordFlow(_ config: [String: String], credentials: [String: String]?, _ view: UIViewController) -> Promise<OktaTokenManager> {
         return buildAndPerformTokenRequest(config, additionalParams: credentials)
     }
+    
+    func logout(_ config: [String: String], idToken: String?, view: UIViewController) -> Promise<Void> {
+        return Promise<Void>(in: .background, { resolve, reject, _ in
+            guard let idToken = idToken else {
+                return reject(OktaError.MissingUserIdToken)
+            }
+
+            guard let issuer = config["issuer"],
+                let logoutRedirectUri = config["logoutRedirectUri"] else {
+                    return reject(OktaError.MissingConfigurationValues)
+            }
+            
+            self.getMetadataConfig(URL(string: issuer))
+                .then { oidConfig in
+
+                    let request = OIDEndSessionRequest(
+                            configuration: oidConfig,
+                            idTokenHint: idToken,
+                            postLogoutRedirectURL: URL(string: logoutRedirectUri)!,
+                            additionalParameters: nil
+                    )
+                    
+                    let agent = OktaExternalUserAgentIOS(presenting: view)
+                    
+                    // Present the logout flow
+                    
+                    OktaAuth.currentAuthorizationFlow = OIDAuthorizationService.present(request, externalUserAgent: agent)
+                    { response, responseError in
+
+                        if let responseError = responseError {
+                            return reject(OktaError.APIError("Logout Error: \(responseError.localizedDescription)"))
+                        }
+
+                        self.clearAuthState()
```

Comment:
I'd prefer not to do this within the `logout` functionality. Instead, we need documentation around the `logout` method to show developers how to:

1) End the user's Okta session in the browser (what this PR adds)
2) Revoke the existing `accessToken` and `refreshToken`
3) Clear the authentication state

Reply:
I think we need two methods: `signOutFromOkta()` and `signOutLocally()`
I've captured this in https://github.com/okta/okta-sdk-appauth-ios/issues/89.

And I agree documentation is needed 👍

Reply:
@nbarbettini Should I implement this in an another PR? Or update existing?

Reply:
Let's do it in this PR.

Reply:
Implemented signOutLocally. Added documentation.